quint/src/effects/modeChecker.ts

```diff
@@ -65,27 +65,49 @@ export class ModeChecker implements IRVisitor {
 
     if (generalMode === mode) {
       this.errors.set(def.id, {
-        location: `Checking modes for ${definitionToString(def)}`,
-        message: `Expected ${mode} mode, found: ${qualifierToString(def.qualifier)}`,
-        children: [],
+        code: 'QNT200',
+        message: `${qualifierToString(def.qualifier)} operators ${modeConstraint(def.qualifier)}, but operator \`${def.name}\` ${this.modeFinderVisitor.explanation}. Use ${qualifierToString(mode)} instead.`,
+        data: { fix: { kind: 'replace', original: qualifierToString(def.qualifier), replacement: qualifierToString(mode) } },
```

Comment:
Since we now have these unique IDs that need to be given to every distinct kind of error, I think it would be a good idea to gather them all into one file. Otherwise I don't see how we will avoid conflict between error IDs, nor will we have a way of surveying all the error we report.

At minimum, I think this would mean defining a union type to enumerate all of the error IDs. But even better IMO would be functions that construct the relevant error given the needed params, tho i can see how this might get a bit unwieldy.

Reply:
We have some error enumeration at ADR 2, but I don't think that will scale well. I have an idea from your suggestions: we add a `ErrorCode` type with enumerated values `'QNT000' | 'QNT001' | ...` and add docstrings to document each of the possibilities. This way, we have some enforcement without adding a big overhead of constructors. WDYT?

Reply:
sounds good!